lua/lvim/core/log.lua

```diff
@@ -36,21 +36,28 @@ function Log:init()
             { level = structlog.formatters.FormatColorizer.color_level() }
           ),
         }),
-        structlog.sinks.File(Log.levels.TRACE, logfile, {
-          processors = {
-            structlog.processors.Namer(),
-            structlog.processors.StackWriter({ "line", "file" }, { max_parents = 3, stack_level = 2 }),
-            structlog.processors.Timestamper "%H:%M:%S",
-          },
-          formatter = structlog.formatters.Format( --
-            "%s [%-5s] %s: %-30s",
-            { "timestamp", "level", "logger_name", "msg" }
-          ),
-        }),
       },
     },
   }
 
+  -- if the cache dir doesn't exist, the file sink causes a crash
+  if vim.loop.fs_stat(get_cache_dir()) then
```

Comment:
Can you show me how this error was getting triggered? The installer should be taking care of this 
https://github.com/LunarVim/LunarVim/blob/e1b8eb01a2e2ca092f761b25777ba60d76609c30/utils/installer/install.sh#L292-L302

Reply:
You're assuming that it is installed using the installer, which may not be the case. For example, while testing the directories are changed _after_ installation. What if the environment variables are changed after the installation?

It's also possible that a user might delete it, or simply not transfer it during a backup, because it's the cache directory. There's also little overhead to this check, there's actually more overhead for doing a pcall instead.

Reply:
It generally sounds like a problem that structlog should handle, but otherwise it's easier and cleaner to just add this line in 
```lua
function Log:init()
  local status_ok, structlog = pcall(require, "structlog")
  if not status_ok then
    return nil
  end
  vim.fn.mkdir(get_lvim_cache_dir(), "p")
```
If you want to be really strict about it, then you can add
```lua
  local utils = require "lvim.utils"
  local cache_dir = get_cache_dir()
  if not utils.is_directory(cache_dir) then
	vim.fn.mkdir(cache_dir, "p")
  end
```